### PR TITLE
Make stitching script more pythonic

### PR DESCRIPTION
**Pull request by ianare**
### This pullrequest changes

Improve the overall readability and maintainability of the Python example stitching script.

Specifically:
* PEP8 fixes
* Removed unused variables
* Start splitting main function into easier to understand functions.
* Use dictionaries for option parsing rather than `if`, this also allows easily showing all available options in the help output.
* Update help text (copied from C++ example)
* Added the AKAZ and BRISK feature finders as options.

### CODE REVIEW

File: samples/python/stitching_detailed.py [hunk 1]
```diff
@@ -8,55 +8,273 @@
...
+)
+parser.add_argument(
+    '--output', action='store', default='result.jpg',
+    help="The default is 'result.jpg'",
+    type=str, dest='output'
+)
+parser.add_argument(
+    '--timelapse', action='store', default=None,
+    help="Output warped images separately as frames of a time lapse movie, "
+         "with "fixed_" prepended to input file names.",
```
Comment (mshabunin) [inline samples/python/stitching_detailed.py]: Non-escaped quotes here.

Comment (ianare) [reply samples/python/stitching_detailed.py]: Thanks, force pushed some fixes.

**Comment by asmorkalov** [comment]
@mshabunin @alalek Do you have any concerns about the patch?

**Comment by ianarco** [comment]
Hi again, any chance of this being merged? Thanks.

**Pull request merged**